modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

```diff
@@ -928,11 +963,27 @@ def date_is_valid?(date, property)
         end
       end
 
-      def raise_date_error(date, property)
-        raise ::Common::Exceptions::UnprocessableEntity.new(
-          detail: "#{date} is not a valid date for #{property}."
+      def raise_date_error(date, property = '/')
+        collect_error_messages(
+          detail: "#{date} is not a valid date.",
+          source: property
         )
       end
+
+      def errors_array
+        @errors ||= []
+      end
+
+      def collect_error_messages(detail: 'Missing or invalid attribute', source: '/',
+                                 title: 'Unprocessable Entity', status: 422.to_s)
+        errors_array.push({ detail:, source:, title:, status: })
+      end
+
+      def raise_error_collection
+        return if errors_array.nil?
```

Comment:
rails console shows an empty array will hit this as false.... 

```
3.2.2 :003 > [].nil?
 => false
```

and errorsy array is instantiated as an empty array. 🤔

Reply:
should we move this check up? 

```
raise_error_collection unless @errors.empty?
```

Reply:
yes that is the change needed there.